<a href="https://colab.research.google.com/github/Sebastian4524/Datavisualization/blob/main/pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **1. Library**

In [ ]:
import numpy as np
import pandas as pd

# **2. Import a dataset**

In [ ]:
s = "/datosp.xlsx"
d = pd.read_excel(s)

# **3. Exploration**

In [ ]:
d.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 759 entries, 0 to 758
Data columns (total 16 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   Region                                        759 non-null    object 
 1   Country code                                  759 non-null    object 
 2   Economy                                       759 non-null    object 
 3   Reporting year                                759 non-null    int64  
 4   Survey name                                   759 non-null    object 
 5   Survey year                                   759 non-null    int64  
 6   Survey coverage                               755 non-null    object 
 7   Welfare type                                  755 non-null    object 
 8   Survey comparability                          755 non-null    float64
 9   Monetary (%)                                  759 non-null    flo

In [ ]:
d.head()

Region Country code  Economy  Reporting year Survey name  Survey year  \
0    SSA          AGO   Angola            2018       IDREA         2018   
1    ECA          ALB  Albania            2012        LSMS         2012   
2    ECA          ALB  Albania            2016         HBS         2016   
3    ECA          ALB  Albania            2017         HBS         2017   
4    ECA          ALB  Albania            2018         HBS         2018   

  Survey coverage Welfare type  Survey comparability  Monetary (%)  \
0               N            c                   2.0     31.122005   
1               N            c                   1.0      0.620691   
2               N            c                   2.0      0.139303   
3               N            c                   2.0      0.392570   
4               N            c                   3.0      0.048107   

   Educational attainment (%) Educational enrollment (%) Electricity (%)  \
0                   29.753423                   27.44306       52.639532   
1                    2.271381                   6.731328        0.496704   
2                    0.622464                          -        0.431298   
3                    0.331231                          -        0.131322   
4                    0.192380                          -         0.06025   

  Sanitation (%) Drinking water (%)  \
0      53.637516          32.106507   
1       3.040233           0.342548   
2       7.603243           8.402681   
3       6.976078            9.27045   
4       6.579772           9.594966   

   Multidimensional poverty headcount ratio (%)  
0                                     47.203606  
1                                      0.792693  
2                                      0.781341  
3                                      0.711346  
4                                      0.293161

In [ ]:
d.tail()

Region Country code       Economy  Reporting year Survey name  \
754    MNA          YEM   Yemen, Rep.            2014         HBS   
755    SSA          ZAF  South Africa            2014         LCS   
756    SSA          ZMB        Zambia            2015    LCMS-VII   
757    SSA          ZWE      Zimbabwe            2017       PICES   
758    SSA          ZWE      Zimbabwe            2019       PICES   

     Survey year Survey coverage Welfare type  Survey comparability  \
754         2014               N            c                   1.0   
755         2014               N            c                   3.0   
756         2015               N            c                   5.0   
757         2017               N            c                   0.0   
758         2019               N            c                   0.0   

     Monetary (%)  Educational attainment (%) Educational enrollment (%)  \
754     19.802757                   15.950564                  44.500461   
755     20.501289                    2.263442                   2.338686   
756     60.793096                   24.374674                    30.3689   
757     34.409153                    6.893227                   3.100654   
758     39.754530                    0.927010                    5.98423   

    Electricity (%) Sanitation (%) Drinking water (%)  \
754       33.891296      41.205913           14.01668   
755        4.093847      35.171476          10.361161   
756       69.213849       59.97659          34.368798   
757       37.062222      67.637062          22.405361   
758        37.99479       38.28523          19.260145   

     Multidimensional poverty headcount ratio (%)  
754                                     35.411829  
755                                     21.710479  
756                                     66.403395  
757                                     39.897895  
758                                     42.397930

# **4. Cleaning**

## 4.1. Column names

In [ ]:
d.rename(columns = {"Reporting year":"year"}, inplace = True)
d.rename(columns = {"Region":"region"}, inplace = True)
d.rename(columns = {"Survey comparability":"survey comparability"}, inplace = True)
d.rename(columns = {"Monetary (%)":"monetary"}, inplace = True)
d.rename(columns = {"Educational attainment (%)":"ea"}, inplace = True)
d.rename(columns = {"Educational enrollment (%)":"ee"}, inplace = True)
d.rename(columns = {"Electricity (%)":"electricity"}, inplace = True)
d.rename(columns = {"Sanitation (%)":"sanitation"}, inplace = True)
d.rename(columns = {"Drinking water (%)":"dwater"}, inplace = True)
d.rename(columns = {"Multidimensional poverty headcount ratio (%)":"multidimensional ratio"}, inplace = True)

In [ ]:
d2=d

## 4.2. Delete columns

In [ ]:
d2 = d2.drop(columns = ["Survey coverage","Welfare type","Survey name","survey comparability"])

In [ ]:
d2.head()

region Country code  Economy  year  Survey year   monetary         ea  \
0    SSA          AGO   Angola  2018         2018  31.122005  29.753423   
1    ECA          ALB  Albania  2012         2012   0.620691   2.271381   
2    ECA          ALB  Albania  2016         2016   0.139303   0.622464   
3    ECA          ALB  Albania  2017         2017   0.392570   0.331231   
4    ECA          ALB  Albania  2018         2018   0.048107   0.192380   

         ee electricity sanitation     dwater  multidimensional ratio  
0  27.44306   52.639532  53.637516  32.106507               47.203606  
1  6.731328    0.496704   3.040233   0.342548                0.792693  
2         -    0.431298   7.603243   8.402681                0.781341  
3         -    0.131322   6.976078    9.27045                0.711346  
4         -     0.06025   6.579772   9.594966                0.293161

## 4.3. Drop NA

In [ ]:
d2 = d2.replace("-",np.NaN)
d2.head()

region Country code  Economy  year  Survey year   monetary         ea  \
0    SSA          AGO   Angola  2018         2018  31.122005  29.753423   
1    ECA          ALB  Albania  2012         2012   0.620691   2.271381   
2    ECA          ALB  Albania  2016         2016   0.139303   0.622464   
3    ECA          ALB  Albania  2017         2017   0.392570   0.331231   
4    ECA          ALB  Albania  2018         2018   0.048107   0.192380   

          ee  electricity  sanitation     dwater  multidimensional ratio  
0  27.443060    52.639532   53.637516  32.106507               47.203606  
1   6.731328     0.496704    3.040233   0.342548                0.792693  
2        NaN     0.431298    7.603243   8.402681                0.781341  
3        NaN     0.131322    6.976078   9.270450                0.711346  
4        NaN     0.060250    6.579772   9.594966                0.293161

In [ ]:
d2 =d2.dropna()

In [ ]:
d2.head()

region Country code    Economy  year  Survey year   monetary         ea  \
0    SSA          AGO     Angola  2018         2018  31.122005  29.753423   
1    ECA          ALB    Albania  2012         2012   0.620691   2.271381   
5    LAC          ARG  Argentina  2010         2010   0.749329   1.904169   
6    LAC          ARG  Argentina  2011         2011   0.707419   1.807899   
7    LAC          ARG  Argentina  2012         2012   0.640789   1.650053   

          ee  electricity  sanitation     dwater  multidimensional ratio  
0  27.443060    52.639532   53.637516  32.106507               47.203606  
1   6.731328     0.496704    3.040233   0.342548                0.792693  
5   0.963598     0.000000    0.683219   0.325656                0.772191  
6   1.051406     0.000000    0.716557   0.308643                0.743913  
7   0.833707     0.000000    0.523555   0.324075                0.675336

# **5. Indexing**

In [ ]:
d2 = d2.set_index(["Economy","year"])
d2.head()

region Country code  Survey year   monetary         ea  \
Economy   year                                                          
Angola    2018    SSA          AGO         2018  31.122005  29.753423   
Albania   2012    ECA          ALB         2012   0.620691   2.271381   
Argentina 2010    LAC          ARG         2010   0.749329   1.904169   
          2011    LAC          ARG         2011   0.707419   1.807899   
          2012    LAC          ARG         2012   0.640789   1.650053   

                       ee  electricity  sanitation     dwater  \
Economy   year                                                  
Angola    2018  27.443060    52.639532   53.637516  32.106507   
Albania   2012   6.731328     0.496704    3.040233   0.342548   
Argentina 2010   0.963598     0.000000    0.683219   0.325656   
          2011   1.051406     0.000000    0.716557   0.308643   
          2012   0.833707     0.000000    0.523555   0.324075   

                multidimensional ratio  
Economy   year                          
Angola    2018               47.203606  
Albania   2012                0.792693  
Argentina 2010                0.772191  
          2011                0.743913  
          2012                0.675336

# **6. Descriptive Statistics**

In [ ]:
# Basic
d2[["ea","ee","electricity","sanitation","dwater","multidimensional ratio"]].describe()

ea          ee  electricity  sanitation      dwater  \
count  336.000000  336.000000   336.000000  336.000000  336.000000   
mean    13.554299    6.946490    12.455075   20.977328    9.780775   
std     15.986560   10.372906    22.527316   24.115456   11.667525   
min      0.000000    0.055241     0.000000    0.000000    0.000000   
25%      3.224699    1.297253     0.231926    3.212466    1.562115   
50%      6.302084    2.846354     1.599842   11.082474    5.690938   
75%     17.102751    6.311659     9.913953   29.749310   12.326958   
max     75.149274   58.042753    97.250843   95.897704   69.206844   

       multidimensional ratio  
count              336.000000  
mean                14.225384  
std                 20.937838  
min                  0.000000  
25%                  1.327811  
50%                  5.093471  
75%                 16.907301  
max                 85.678405

In [ ]:
import pandas as pd

# Define una función personalizada para el formato
def custom_format(x):
    if isinstance(x, float):
        return '{:.3f}'.format(x)
    return str(x)

# Aplica la función de formato personalizada a todos los datos del DataFrame
pd.options.display.float_format = custom_format

# Ejecuta tu código
d2[["ea", "ee", "electricity", "sanitation", "dwater", "multidimensional ratio"]].describe()

ea      ee  electricity  sanitation  dwater  multidimensional ratio
count 336.000 336.000      336.000     336.000 336.000                 336.000
mean   13.554   6.946       12.455      20.977   9.781                  14.225
std    15.987  10.373       22.527      24.115  11.668                  20.938
min     0.000   0.055        0.000       0.000   0.000                   0.000
25%     3.225   1.297        0.232       3.212   1.562                   1.328
50%     6.302   2.846        1.600      11.082   5.691                   5.093
75%    17.103   6.312        9.914      29.749  12.327                  16.907
max    75.149  58.043       97.251      95.898  69.207                  85.678

In [ ]:
d2[["ea", "ee" , "sanitation", "dwater", "multidimensional ratio"]].describe().applymap(lambda x:"{:,.1f}".format(x))

ea     ee sanitation dwater multidimensional ratio
count  336.0  336.0      336.0  336.0                  336.0
mean    13.6    6.9       21.0    9.8                   14.2
std     16.0   10.4       24.1   11.7                   20.9
min      0.0    0.1        0.0    0.0                    0.0
25%      3.2    1.3        3.2    1.6                    1.3
50%      6.3    2.8       11.1    5.7                    5.1
75%     17.1    6.3       29.7   12.3                   16.9
max     75.1   58.0       95.9   69.2                   85.7

# **7. Conclusions**

The country that has the highest educational attainment is 75.1 % meaning that just 24.9% of their population does not have a degree
